packages/swarm-components/src/Textarea.jsx

```diff
@@ -64,22 +57,25 @@ class Textarea extends React.Component<Props, State> {
         const {
             maxLength,
             autosize, // eslint-disable-line no-unused-vars
+            value = '',
             ...other
         } = this.props;
 
-        const remainingCharacters = (parseInt(maxLength, 10) - getCharacterCount(this.props.value));
-        const textareaStatus = getTextareaStatus({ ...this.props, error: this.props.error || remainingCharacters < 0 });
+        const charLength = value.length;
+        const remainingCharacters = getRemainingCharacters(maxLength, charLength);
```

Comment:
I don't love this solution because we're computing the remaining characters here, and then again within the CharCount component. What I hadn't realized before is that the remainingCharacters is required to determine error states on the form field itself.

Options:
- Keep as is.
- Pass `remainingCharacters` as a prop to CharCount (original implementation).
- Keep as is, but add `remainingCharacters` to the component state of the textarea and provide a callback as a prop to `CharCount` to update it. All computation done within CharCount component. 

Don't particularly like any of these. Thoughts?

Reply:
an `onChange` callback might be a nice solution - something that just gets called with the value of 'remaining characters'. You could even/alternatively have an `onError` callback that just gets fired when there are less than 0 characters remaining. Either solution leaves the question of what to do with the 'event' up to the parent.

That said, looking at the logic, a `getRemainingCharacters` that gets passed the current value and a max value is a really trivial abstraction of "is this number bigger than this number", so you probably don't need it - you're getting a value from subtracting the numbers, and then checking whether that is less than zero, instead of just `charLength > maxLength`. Probably best not to overthink it...

Reply:
Either keep it as is or go with the original implementation. As @mmcgahan said, let's not overthink it. The important part is that this is solved within Textarea / TextInput and derived from props that make sense.

Reply:
Ok, you're both recommending different things, so I'm going to go back to the original implementation and then merge this. If we need something different later, it can be changed.

Reply:
be empowered, Sadaf!